algo/src/main/java/org/neo4j/graphalgo/linkprediction/LinkPrediction.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2017 "Neo4j, Inc." <http://neo4j.com>
+ * <p>
+ * This file is part of Neo4j Graph Algorithms <http://github.com/neo4j-contrib/neo4j-graph-algorithms>.
+ * <p>
+ * Neo4j Graph Algorithms is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.graphalgo.linkprediction;
+
+import org.neo4j.graphalgo.core.ProcedureConfiguration;
+import org.neo4j.graphdb.*;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.UserFunction;
+
+import java.util.*;
+
+public class LinkPrediction {
+
+    @Context
+    public GraphDatabaseAPI api;
+
+    @UserFunction("algo.linkprediction.adamicAdar")
```

Comment:
do you plan this also as a proc on the algo-graph?
then we could use intersection etc.

Reply:
Wasn't planning to as this one and resource allocation are used to find closeness between pairs of nodes. I don't think you'd use them to do all pairs similarity like we do with the similarity algos (Pearson, Cosine, Jaccard, etc)